test/data/test_builtin_datasets.py

```diff
@@ -56,11 +46,26 @@ def test_wikitext2(self):
         self.assertEqual(len(train_dataset), 2049990)
         self.assertEqual(len(test_dataset), 241859)
         self.assertEqual(len(valid_dataset), 214417)
+        self.assertEqual(train_dataset[20:25], torch.tensor([5024, 89, 21, 3, 1838]).long())
+        self.assertEqual(test_dataset[30:35], torch.tensor([914, 4, 36, 11, 569]).long())
+        self.assertEqual(valid_dataset[40:45], torch.tensor([925, 8, 2, 150, 8575]).long())
 
         vocab = train_dataset.get_vocab()
         tokens_ids = [vocab[token] for token in 'the player characters rest'.split()]
         self.assertEqual(tokens_ids, [2, 286, 503, 700])
 
+        # Add test for the subset of the standard datasets
+        train_dataset, test_dataset = torchtext.experimental.datasets.raw.WikiText2(data_select=('train', 'test'))
+        next(iter(train_dataset))
+        self.assertEqual(next(iter(train_dataset)), ' = Valkyria Chronicles III = \n')
+        next(iter(test_dataset))
```

Comment:
based on offline discussion: Another way to get rid of the iterator that's being purposefully overwritten by this naming is to use ```del train_iter```